### ISSUE TITLE
colors initially not working with bat and default config

### ISSUE THREAD

**Issue by Flupp** [Morganamilo/paru#673]
### Affected Version

paru v1.9.2 - libalpm v13.0.1

### Description

With the default config, when `bat` is installed, `less` does not show colors but visualizes the escape sequences instead during review.

I worked around the problem by adding the following to `paru.conf`:
```ini
[bin]
Pager = /usr/bin/less --RAW-CONTROL-CHARS
```

Is this intended?

### Output

```
ESC[1m/home/paru/.cache/paru/clone/paru/PKGBUILDESC[0m
ESC[38;2;124;120;101m#ESC[0mESC[38;2;124;120;101m Maintainer: Morgan <morganamilo@archlinux.org>ESC[0m
ESC[38;2;255;255;255mpkgnameESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;230;219;116mparuESC[0m
ESC[38;2;255;255;255mpkgverESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;230;219;116m1.9.2ESC[0m
ESC[38;2;255;255;255mpkgrelESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;230;219;116m1ESC[0m
ESC[38;2;255;255;255mpkgdescESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;255;255;255m'ESC[0mESC[38;2;230;219;116mFeature packed AUR helperESC[0mESC[38;2;255;255;255m'ESC[0m
ESC[38;2;255;255;255murlESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;255;255;255m'ESC[0mESC[38;2;230;219;116mhttps://github.com/morganamilo/paruESC[0mESC[38;2;255;255;255m'ESC[0m
ESC[38;2;255;255;255msourceESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;255;255;255m(ESC[0mESC[38;2;255;255;255m"ESC[0mESC[38;2;255;255;255m$ESC[0mESC[38;2;255;255;255mpkgnameESC[0mESC[38;2;230;219;116m-ESC[0mESC[38;2;255;255;255m$ESC[0mESC[38;2;255;255;255mpkgverESC[0mESC[38;2;230;219;116m.tar.gz::https://github.com/Morganamilo/paru/archive/vESC[0mESC[38;2;255;255;255m$ESC[0mESC[38;2;255;255;255mpkgverESC[0mESC[38;2;230;219;116m.tar.gzESC[0mESC[38;2;255;255;255m"ESC[0mESC[38;2;255;255;255m)ESC[0m
ESC[38;2;255;255;255mbackupESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;255;255;255m(ESC[0mESC[38;2;255;255;255m"ESC[0mESC[38;2;230;219;116metc/paru.confESC[0mESC[38;2;255;255;255m"ESC[0mESC[38;2;255;255;255m)ESC[0m
ESC[38;2;255;255;255marchESC[0mESC[38;2;249;38;114m=ESC[0mESC[38;2;255;255;255m(ESC[0mESC[38;2;255;255;255m'ESC[0mESC[38;2;230;219;116mi686ESC[0mESC[38;2;255;255;255m'ESC[0mESC[38;2;255;255;255m ESC[0mESC[3
[…]
```

`paru.conf`:
```ini
#
# $PARU_CONF
# /etc/paru.conf
# ~/.config/paru/paru.conf
#
# See the paru.conf(5) manpage for options

#
# GENERAL OPTIONS
#
[options]
PgpFetch
Devel
Provides
DevelSuffixes = -git -cvs -svn -bzr -darcs -always -hg
#AurOnly
#BottomUp
#RemoveMake
#SudoLoop
#UseAsk
#SaveChanges
#CombinedUpgrade
#CleanAfter
#UpgradeMenu
#NewsOnUpgrade

#LocalRepo
#Chroot
#Sign
#SignDb

#
# Binary OPTIONS
#
#[bin]
#FileManager = vifm
#MFlags = --skippgpcheck
#Sudo = doas
```

`pacman.conf`:
```ini
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = auto

# Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
#IgnorePkg   =
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
#Color
Color
#NoProgressBar
CheckSpace
#VerbosePkgLists
VerbosePkgLists
#ParallelDownloads = 5

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Arch Linux
# packagers with `pacman-key --populate archlinux`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

#[testing]
#Include = /etc/pacman.d/mirrorlist

[core]
Include = /etc/pacman.d/mirrorlist

[extra]
Include = /etc/pacman.d/mirrorlist

#[community-testing]
#Include = /etc/pacman.d/mirrorlist

[community]
Include = /etc/pacman.d/mirrorlist

# If you want to run 32 bit applications on your x86_64 system,
# enable the multilib repositories as required here.

#[multilib-testing]
#Include = /etc/pacman.d/mirrorlist

#[multilib]
#Include = /etc/pacman.d/mirrorlist

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs
```

**Comment by Morganamilo** [comment]
Do you have the LESS variable set?

**Comment by Morganamilo** [comment]
You need --RAW-CONTROL-CHARS in your less flags.

**Comment by Flupp** [comment]
Adding `--RAW-CONTROL-CHARS` to the `LESS` environment variable works. However, I find it a bit counter-intuitive that it works with an unset `LESS` variable, but when it is set, then it is necessary to add `--RAW-CONTROL-CHARS`. Also, it might get complicated if other tools add other constraints to a set `LESS` variable.

**Comment by Morganamilo** [comment]
Without LESS set paru will use some defaults. With it set it will use what was set. I believe git does the same with 'git diff' and such.

**Comment by Flupp** [comment]
Hmm, I see:

https://github.com/Morganamilo/paru/blob/4d2699ae38efa8536b7e7a8d7f07974084f70ab3/src/install.rs#L776-L778

Maybe, instead of using the `LESS` variable (or in addition to it), pass `--RAW-CONTROL-CHARS` directly on the command line:

https://github.com/Morganamilo/paru/blob/4d2699ae38efa8536b7e7a8d7f07974084f70ab3/src/install.rs#L766-L771

So maybe replace line 771 above by `.unwrap_or_else(|| "less --RAW-CONTROL-CHARS".to_string());`. Arguments from the command line even take precedence over arguments from the `LESS` variable. So this should work even if `LESS=-+R`.

But probably there is a catch I don’t see. :)

**Comment by Morganamilo** [comment]
I'd rather not do this. Passing on the command line will fail if the user is not using less as their pager. Plus like I said git does it this way so you're still going to end up with a broken git diff.

**Comment by Flupp** [comment]
Note that I just suggested to add the option to the default command, i.e., it does not apply when a pager is explicitly configured via the config, `PARU_PAGER`, or `PAGER`.

Anyway, you are right, Git has the same behavior and there I also worked around it by setting the pager in the config. (I did not remember that until now.)

I now understand that there is no perfect solution. So feel free to close this issue.
